packages/mermaid/src/diagrams/sankey/sankeyDB.ts

```diff
@@ -0,0 +1,124 @@
+// import { log } from '../../logger.js';
+// import mermaidAPI from '../../mermaidAPI.js';
+import * as configApi from '../../config.js';
+import common from '../common/common.js';
+import {
+  setAccTitle,
+  getAccTitle,
+  getAccDescription,
+  setAccDescription,
+  setDiagramTitle,
+  getDiagramTitle,
+  clear as commonClear,
+} from '../../commonDb.js';
+
+// export const parseDirective = function (statement, context, type) {
+//   mermaidAPI.parseDirective(this, statement, context, type);
+// };
+
+// export const cleanupComments = (text: string): string => {
+//   return text.trimStart().replace(/^\s*%%(?!{)[^\n]+\n?/gm, '');
+// };
+let links: Array<Link> = [];
+let nodes: Array<Node> = [];
+let nodesHash: Record<string, Node> = {};
+
+const clear = function () {
+  links = [];
+  nodes = [];
+  nodesHash = {};
+  commonClear();
+};
+
+type Nullable<T> = T | null;
+
+// interface ILink {
+//   source?: Node;
+//   target?: Node;
+//   value?: number;
+// }
+
+class Link {
+  source: Nullable<Node>;
+  target: Nullable<Node>;
+  value: Nullable<number>;
+  constructor() {
+    this.source = null;
+    this.target = null;
+    this.value = 0;
+  }
+}
```

Comment:
Please use `undefined` instead of `null`. 
Then you can use the `?` syntax instead of `Nullable`.

Reply:
That will be redesigned

Reply:
Fixed